### ISSUE TITLE
Cannot change secret type. Secret type is immutable

### ISSUE THREAD

**Issue by jstewart612** [1Password/onepassword-operator#111]
### Your environment

Operator Version: 1.4.1

Connect Server Version: 1.5.2

Kubernetes Version: 1.23.6

## What happened?

```
{"level":"info","ts":1651866038.7324877,"logger":"controller_onepassworditem","msg":"Reconciling OnePasswordItem","Request.Namespace":"default","Request.Name":"cloudflare-apikey"}
{"level":"error","ts":1651866038.7860298,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"onepassworditem-controller","request":"default/cloudflare-apikey","error":"Cannot change secret type. Secret type is immutable","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/workspace/vendor/github.com/go-logr/zapr/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:258\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90"}
```

## What did you expect to happen?

It created the secret but then began looping this despite the fact I didn't change anything.

## Steps to reproduce

kubectl apply -f this manifest:
```
apiVersion: onepassword.com/v1
kind: OnePasswordItem
metadata:
  annotations:
    argocd.argoproj.io/sync-wave: "1"
  name: cloudflare-apikey
  namespace: argocd
spec:
  itemPath: vaults/k3s/items/cloudflare-apikey
type: Opaque
```

## Notes & Logs

```
{"level":"info","ts":1651866038.7324877,"logger":"controller_onepassworditem","msg":"Reconciling OnePasswordItem","Request.Namespace":"default","Request.Name":"cloudflare-apikey"}
{"level":"error","ts":1651866038.7860298,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"onepassworditem-controller","request":"default/cloudflare-apikey","error":"Cannot change secret type. Secret type is immutable","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/workspace/vendor/github.com/go-logr/zapr/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:258\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90"}
```

**Comment by slok** [comment]
Hi @jstewart612! It may be related to what #109 fixes.

**Comment by jstewart612** [comment]
@slok I responded to an unanswered review in the linked PR.  Here's to hoping that unblocks the approver's concerns and they move it along :)

**Comment by jpcoenen** [comment]
Thank you for reporting this issue! This should indeed be fixed by #109.
